File: solutions/java/jobs.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class jobs {

    public static void main(String[] args) {

		try {
			if(args.length < 1) {
				System.out.println("syntax: jobs.py [list | show jobid | results jobid | delete jobid]");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);
			if(args[0].equalsIgnoreCase("list")) {
				for (String jobid : service.getJobs().keySet()) {
					Job job = service.getJobs().get(jobid);
					System.out.println("  " + jobid + " [" + job.getEventSearch() + "]");
				}
			}
			else if(args[0].equalsIgnoreCase("show")) {
				Job job = service.getJobs().get(args[1]);
				for (String key : job.keySet())
					  System.out.println("  " + key + " = " + job.get(key));
			}
			else if(args[0].equalsIgnoreCase("results")) {
				Job job = service.getJobs().get(args[1]);
		        InputStream response = job.getResults();
		        int chr = 0;
				while (chr != -1) System.out.write(chr = response.read());
			}
			else if(args[0].equalsIgnoreCase("delete")) {
				Job job = service.getJobs().get(args[1]);
				job = job.cancel();  
				System.out.println("  " + args[1] + " deleted.");
			}
			else {
				throw new Exception("Unknown command \"" + args[0] + "\"");
			}
			
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: solutions/python/jobs.py

import splunklib.client as client
import utils, sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
try: 
    if len(sys.argv) == 1:
        print "syntax: jobs.py [list | show jobid | results jobid | delete jobid]"
        sys.exit()
    
    service = client.connect(**opts.kwargs)

    if sys.argv[1] == "list":
        for job in service.jobs:
            print "  %s [%s]" % (job.sid, job.eventSearch)

    elif sys.argv[1] == "show":
        job = service.jobs[sys.argv[2]]
        for key in sorted(job.state.keys()):
            print '  ' + key
            for sub in job[key]:
                print "    %s: %s" % (sub, job[key][sub]) 

    elif sys.argv[1] == "results":
        job = service.jobs[sys.argv[2]]        
        if not job.is_done():
            raise Exception("Job not complete")
        response = job.results()
        print response
        
    elif sys.argv[1] == "delete":
        job = service.jobs[sys.argv[2]]
        job.cancel();
        print "  job %s deleted" % job.sid
 
    else:
        raise Exception("unknown command \"%s\"" % sys.argv[1])

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
